microprofile/testing/junit5/src/main/java/io/helidon/microprofile/testing/junit5/HelidonJunitExtension.java

```diff
@@ -234,6 +242,13 @@ private void validatePerClass() {
                 }
             }
         }
+
+        AddJaxRs addJaxRsAnnotation = testClass.getAnnotation(AddJaxRs.class);
+        if (addJaxRsAnnotation != null){
+            if (testClass.getAnnotation(DisableDiscovery.class) == null){
+                throw new RuntimeException("@AddJaxRs annotation should be used only with @DisableDiscovery annotation.");
```

Comment:
Also check if one of the extensions added automatically by this annotation isn't there added manually eg: `@AddExtension(JaxRsCdiExtension.class)` etc...